server.js

```diff
@@ -190,18 +199,20 @@ async function getSessionId(account_id, tag) {
  */
 async function createParticipant(account_id, tag) {
   // create a participant for this browser user
-  var participantBody = new BandwidthWebRTC.Participant({
+  var participantBody = {
     tag: tag,
     publishPermissions: ["AUDIO"],
-  });
+    callbackUrl: "foo",
```

Comment:
I believe there's an issue either in the WebRTC SDK or the backend implementation. As it exists now this call will fail if you don't supply a callback URL, which is an optional field. The response fails to parse with this error:

```
Given value: null
Type: 'object'
Expected type: 'Optional<string>'
Path: participant › callbackUrl
    at DefaultRequestBuilder.callAsJson (/Users/carrington/workspace/node-webrtc/dist/webrtc.cjs.development.js:1091:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async createParticipant (file:///Users/carrington/workspace/webrtc-hello-world-js/server.js:209:26)
    at async file:///Users/carrington/workspace/webrtc-hello-world-js/server.js:52:32 {
```

This is what the 

```
{
    "participant": {
        "callbackUrl": null,
        "deviceApiVersion": "V2",
        "id": "f2722597-7297-4c28-94df-ba5aa604d936",
        "publishPermissions": [
            "VIDEO",
            "AUDIO"
        ],
        "sessions": [],
        "subscriptions": {},
        "tag": "participant2"
    },
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjoiOTkwMTA3OCIsInAiOiJmMjcyMjU5Ny03Mjk3LTRjMjgtOTRkZi1iYTVhYTYwNGQ5MzYiLCJ2IjoiVjIiLCJleHAiOjE2MTk4NTA1OTcsInRpZCI6IisyNjE4ODk4MDc2MjI1NTYifQ.FfpHA6o0BOYZf4ZtwmtTzXwrqK-WODN66kuUL5ikxhk"
}
```

Any thoughts on this?

Reply:
Did you try without a `null` value?  `null` is not the same as optional in JSON and by the OpenAPI standards.  Null requires a field to be `nullable: true` which is different than optional.  Optional means the field does not appear in the JSON object.

Reply:
In JavaScript setting a JS object with a field to `undefined` will make it so that the field does not appear in a JSON serialized body. 

```js
obj = { field1: undefined, field2: null }
console.log( JSON.stringify(obj) )
//  "{"field2": null}"
```

Reply:
That response JSON is what is emitted from webrtc_manager. 

Am I correct that adding `nullable: true` here will fix it?
https://github.com/Bandwidth/api-docs/blob/main/external/specs/WebRtc/openapispec.json#L766

Reply:
I verified that this is fixed (I can omit the callbackUrl parameter) in the beta SDK :)